### PR TITLE
Markdown support for posts

### LINKED ISSUE #100: Markdown support for posts
This would make the life of the WRC a tad easier.

### PR DESCRIPTION
**Pull request by timhabermaas**
This should close #100. Could someone review it?

I'm not sure whether converting old posts to markdown format is desirable. If not, you can simply cherry pick the first commit.

### CODE REVIEW

File: WcaOnRails/app/views/posts/_post.html.erb [hunk 1]
```diff
@@ -22,6 +22,6 @@
     <div>
       <em>Announced by <%= post.author.name %> on <%= local_time(post.created_at) %></em>
     </div>
-    <%= post.body.html_safe %>
```
Comment (jfly) [inline WcaOnRails/app/views/posts/_post.html.erb]: Spacing nit: the convention I believe I've been following is to put a space after the equals sign here. (full disclosure: I'm super new to Rails, and am not married to any such decisions, but do want to be consistent)

**Comment by jfly** [comment]
Thanks, @timhabermaas! Super excited to see someone jump in =)

My gut here is that converting old posts to markdown only has the potential to break things, with no upside. What are your thoughts?

File: WcaOnRails/app/views/posts/_post.html.erb (hunk 1 above)
Comment (timhabermaas) [inline WcaOnRails/app/views/posts/_post.html.erb]: In earlier versions of Rails you needed to explicitly sanitize user input using `h` and Rails' convention used to be `<%=h post.body %>` (can't find old docs right now, but at least this SO answer: http://stackoverflow.com/a/692926), so I simply used that convention for other helpers (e.g. date format). I can change it if you want.

**Comment by timhabermaas** [comment]
Converting old result posts to markdown seems to work fine (have only checked one post, though). You're right, though. It's probably not 100% bulletproof.

Personally I'd convert the posts to markdown because at one point you might want to customize Redcarpet's renderer (e.g. add `title` tags for links which point to a competitor, disallow headings, ...) and you can't do this with raw HTML still hanging around. Another point: Currently Redcarpet doesn't escape HTML tags, but IMO it's reasonable for a Markdown renderer to convert `<a>` to `&lt;a&gt;`. I wouldn't rely on this behavior of the current renderer.

File: WcaOnRails/app/views/posts/_post.html.erb (hunk 1 above)
Comment (jfly) [inline WcaOnRails/app/views/posts/_post.html.erb]: Haha, wow. That's funny. This example code looks kind of schizophrenic: http://guides.rubyonrails.org/i18n.html#adding-date-time-formats. Leaving this as is seems fine to me.

**Comment by jfly** [comment]
Converting to markdown sounds fine, then. Consistency sounds nicer, and it's not a big deal if some stuff breaks. I'll accept this PR, but I won't be able to push this to production right now. Going to be travelling for the next day or so. If someone else (@lgarron) wants to push this, then they should feel free, otherwise I'll try to get to this tomorrow.

File: WcaOnRails/app/views/posts/_post.html.erb (hunk 1 above)
Comment (timhabermaas) [inline WcaOnRails/app/views/posts/_post.html.erb]: Haha, dat inconsistency...

**Comment by jfly** [comment]
Oh, silly me. We shouldn't bother converting all old html to markdown until we've ported our announcement code to produce markdown. I'm actually in the process of porting our announcement code from php to rails right now, so I'll take this task on.

File: WcaOnRails/app/views/posts/_post.html.erb (hunk 1 above)
Comment (jfly) [inline WcaOnRails/app/views/posts/_post.html.erb]: Just ran this script on the live site and it worked perfectly =). Thanks again, @timhabermaas!

**Pull request merged**